### PR TITLE
API Updates

### PR DESCRIPTION
**Pull request by andrewhart098**
This PR adds the ability to reach two new endpoints introduced in QC API v2.

The first of the endpoints allows users to download data they have purchased from QuantConnect.  The data is stored to disk in the same folder-hierarchy as Lean uses to store data.

The second endpoint allows users to read logs of live running algorithms.

### CODE REVIEW

File: Api/Api.cs [hunk 1]
```diff
@@ -358,6 +432,28 @@ public virtual IEnumerable<MarketHoursSegment> MarketToday(DateTime time, Symbol
...
+        /// <returns>Path that specifies where data will be downloaded</returns>
+        private string CreatePathForDownloadedData(Symbol symbol, Resolution resolution, DateTime date)
+        {
+            string path;
+
+            if (resolution == Resolution.Daily || resolution == Resolution.Hour)
+                path = Config.Get("data-folder") + symbol.ID.SecurityType.ToLower() + "/" + symbol.ID.Market.ToLower() +
+                       "/" + resolution.ToLower() + "/" + symbol.Value.ToLower() + ".zip";
+            else
+                path = Config.Get("data-folder") + symbol.ID.SecurityType.ToLower() + "/" + symbol.ID.Market.ToLower() +
```
Comment (jaredbroad) [inline Api/Api.cs]: Please pass in a "data-folder" variable in the constructor, and save it to the class, so its not loading from config each time.

File: Common/Interfaces/IApi.cs [hunk 2]
```diff
@@ -132,7 +132,32 @@ public interface IApi : IDisposable
...
         LiveList LiveList();
-        
+
+        /// <summary>
+        /// Gets the logs of a specific live algorithm 
+        /// </summary>
+        /// <param name="projectId">Project Id of the live running algorithm</param>
+        /// <param name="algorithmId">Algorithm Id of the live running algorithm</param>
+        /// <returns>List of strings that represent the logs of the algorithm</returns>
+        LiveLog ReadLiveLogs(int projectId, string algorithmId);
```
Comment (jaredbroad) [inline Common/Interfaces/IApi.cs]: Please pass in a start, end and format specifier:

format - raw or json
start (api takes timestamp start)
end (api takes timestamp end)

The C# method should have a datetime input which is converted to -> timestamp in unix sec. Defaulting the start to 0 and end to unix-now.

File: Api/Api.cs (hunk 1 above)
Comment (andrewhart098) [inline Api/Api.cs]: There is no constructor for the Api class.  I will instead pass it in to the Initialize() method and save it off there. I think this is what you meant.

File: Api/Api.cs [hunk 3]
```diff
@@ -443,17 +453,28 @@ private string CreatePathForDownloadedData(Symbol symbol, Resolution resolution,
             string path;
 
             if (resolution == Resolution.Daily || resolution == Resolution.Hour)
-                path = Config.Get("data-folder") + symbol.ID.SecurityType.ToLower() + "/" + symbol.ID.Market.ToLower() +
+                path = _dataFolder + symbol.ID.SecurityType.ToLower() + "/" + symbol.ID.Market.ToLower() +
                        "/" + resolution.ToLower() + "/" + symbol.Value.ToLower() + ".zip";
             else
-                path = Config.Get("data-folder") + symbol.ID.SecurityType.ToLower() + "/" + symbol.ID.Market.ToLower() +
+                path = _dataFolder + symbol.ID.SecurityType.ToLower() + "/" + symbol.ID.Market.ToLower() +
```
Comment (jaredbroad) [inline Api/Api.cs]: Path.Combine() is a safe cross platform alternative here; or even better use the:
"LeanData.GenerateRelativeZipFilePath" method in the toolbox LEAN Data Writer,

File: Api/Api.cs [hunk 4]
```diff
@@ -443,17 +453,28 @@ private string CreatePathForDownloadedData(Symbol symbol, Resolution resolution,
...
         }
 
         /// <summary>
+        /// Convert C# dateTime to Unix Epoch DateTime
+        /// </summary>
+        /// <param name="dateTime">DateTime to be converted</param>
+        /// <returns>The total number of seconds since Jan 1, 1970</returns>
+        private static long DateTimeToUnixTimestamp(DateTime dateTime)
+        {
+            return (long) (dateTime - new DateTime(1970, 1, 1, 0, 0, 0, 0, System.DateTimeKind.Utc)).TotalSeconds;
```
Comment (jaredbroad) [inline Api/Api.cs]: We have a static method Time.DateTimeToUnixTimeStamp()

File: Tests/API/Api.cs [hunk 5]
```diff
@@ -256,8 +254,8 @@ public void Download_AndSave_DataCorrectly()
             Assert.IsTrue(downloadedMinuteData);
             Assert.IsTrue(downloadedDailyData);
             
-            Assert.IsTrue(File.Exists(Config.Get("data-folder") + "forex/oanda/daily/eurusd.zip"));
-            Assert.IsTrue(File.Exists(Config.Get("data-folder") + "forex/oanda/minute/eurusd/20131011_quote.zip"));
+            Assert.IsTrue(File.Exists(_dataFolder + "forex/oanda/daily/eurusd.zip"));
```
Comment (jaredbroad) [inline Tests/API/Api.cs]: Perhaps its worth deleting the file if it exists already? Otherwise its might read a false positive

File: Api/Api.cs [hunk 6]
```diff
@@ -271,6 +276,85 @@ public LiveList LiveList()
...
+            var path = Path.Combine(_dataFolder, LeanData.GenerateRelativeZipFilePath(symbol.Value, symbol.ID.SecurityType, symbol.ID.Market, date, resolution));
+
+            // Make sure the directory exist before writing
+            (new FileInfo(path)).Directory.Create();
+
+            // Download and save the data
+            var uri     = new Uri(link.DataLink);
+            var client  = new RestClient(uri.Scheme + "://" + uri.Host);
+            var request = new RestRequest(uri.PathAndQuery, Method.GET);
+            client.DownloadData(request).SaveAs(path);
```
Comment (jaredbroad) [inline Api/Api.cs]: Nice

**Pull request merged**